src/main/java/org/opensearch/ad/transport/ValidateAnomalyDetectorTransportAction.java

```diff
@@ -0,0 +1,260 @@
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+public class ValidateAnomalyDetectorTransportAction extends
+    HandledTransportAction<ValidateAnomalyDetectorRequest, ValidateAnomalyDetectorResponse> {
+    private static final Logger logger = LogManager.getLogger(ValidateAnomalyDetectorTransportAction.class);
+
+    private final Client client;
+    private final ClusterService clusterService;
+    private final NamedXContentRegistry xContentRegistry;
+    private final AnomalyDetectionIndices anomalyDetectionIndices;
+    private final SearchFeatureDao searchFeatureDao;
+    private volatile Boolean filterByEnabled;
+
+    @Inject
+    public ValidateAnomalyDetectorTransportAction(
+        Client client,
+        ClusterService clusterService,
+        NamedXContentRegistry xContentRegistry,
+        Settings settings,
+        AnomalyDetectionIndices anomalyDetectionIndices,
+        ActionFilters actionFilters,
+        TransportService transportService,
+        SearchFeatureDao searchFeatureDao
+    ) {
+        super(ValidateAnomalyDetectorAction.NAME, transportService, actionFilters, ValidateAnomalyDetectorRequest::new);
+        this.client = client;
+        this.clusterService = clusterService;
+        this.xContentRegistry = xContentRegistry;
+        this.anomalyDetectionIndices = anomalyDetectionIndices;
+        this.filterByEnabled = AnomalyDetectorSettings.FILTER_BY_BACKEND_ROLES.get(settings);
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(FILTER_BY_BACKEND_ROLES, it -> filterByEnabled = it);
+        this.searchFeatureDao = searchFeatureDao;
+    }
+
+    @Override
+    protected void doExecute(Task task, ValidateAnomalyDetectorRequest request, ActionListener<ValidateAnomalyDetectorResponse> listener) {
+        User user = getUserContext(client);
+        AnomalyDetector anomalyDetector = request.getDetector();
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            resolveUserAndExecute(user, listener, detector -> validateExecute(request, user, context, listener));
+        } catch (Exception e) {
+            logger.error(e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void resolveUserAndExecute(
+        User requestedUser,
+        ActionListener<ValidateAnomalyDetectorResponse> listener,
+        Consumer<AnomalyDetector> function
+    ) {
+        try {
+            // Check if user has backend roles
+            // When filter by is enabled, block users validating detectors who do not have backend roles.
+            if (filterByEnabled && !checkFilterByBackendRoles(requestedUser, listener)) {
+                return;
+            }
+            // Validate Detector
+            function.accept(null);
```

Comment:
Don't need to pass any parameter to consumer function here. How about use `AnomalyDetectorFunction`?

Reply:
Done